### ISSUE TITLE
Reverse restrictToViewport

### ISSUE THREAD

**Issue by liamb13** [justintaddei/v-shared-element#28]
Wondering how it could be possible to work restrictToViewport going the opposite direction.

restrictToViewport works great for: 
/contacts to /contacts/12

But not so much going from
/contacts/12 to /contacts

To replicate, using [your dynamic list example](https://justintaddei.github.io/v-shared-element/examples/dynamic-lists) (obviously with restrictToViewport enabled)
/dynamic-lists to /dynamic-lists/8 works as intended:
![Lists to List works](https://user-images.githubusercontent.com/69778531/124050532-10e29d80-da5e-11eb-95c2-5b1b2055282b.gif)

But the opposite /dynamic-lists/8 to /dynamic-lists doesn't
![List Item to Lists](https://user-images.githubusercontent.com/69778531/124050754-88183180-da5e-11eb-97fa-a684f6ab5dd1.gif)

**Comment by justintaddei** [comment]
That would be a simple check of the `activeElement`'s `DOMRect` at [index.ts:38](https://github.com/justintaddei/v-shared-element/blob/9183a19a4b7f3521c1d62bd0dc5186a364722e1e/src/index.ts#L38)

`restrictToViewport` was originally developed to alleviate the performance cost of having many shared elements on a single page. This also helped with the jarring effect of seeing a shared element, that was not in the viewport on the route you left, suddenly fly into the viewport on the new route. Since `v-shared-element` only considers shared elements that have been cached on the previous route, there is little to gain from disabling those which are not in the viewport on the new route.

What was your main reasoning behind trying to do this? If think it would be a worthwhile feature, I will look into developing it!

**Comment by liamb13** [comment]
So idea being that a user could navigate to a page from google search or their history. Then by clicking the back button there’s a nice interaction to take them to the item in the list.

Does that make sense?

**Comment by justintaddei** [comment]
I apologize, but I'm not quite sure I follow what you're saying. If they arrive at `/examples/dynamic-lists/2` from Google search, and hit the back button, they will be taken back to Google.

> But the opposite /dynamic-lists/8 to /dynamic-lists doesn't  

This gif you provided displays the behavior I intended. What is it that "doesn't work" exactly? Do you mean to have it not animate back to the list because the list item is outside of the viewport?

**Comment by liamb13** [comment]
Sorry, I'm having a hard time describing it. I've simulated the intended effect:
![Intended](https://user-images.githubusercontent.com/69778531/124063517-688d0300-da76-11eb-8b3b-778bfa47001c.gif)

So the viewport would be where that item appears on the page after the transition.

**Comment by justintaddei** [comment]
Oh, so you're suggesting that the page is scrolled so that the active shared-element on the incoming route is positioned inside the viewport?

**Comment by liamb13** [comment]
Yep, spot on! Sorry I was a bit over the shop trying to explain. I think as an option might be best because I can see why some would prefer as is.

**Comment by justintaddei** [comment]
That's a cool idea. I'll look into implementing it!

**Comment by 719media** [comment]
If the route is changing between, this is generally best solved by using <keep-alive> along with vue-router in vue, which will take care of scrolling for you.

**Comment by justintaddei** [comment]
@719media I believe the difference they are talking about is for when navigating to pages that have not yet been visited. Where vue-router wouldn't work.
